src/Filesystem.cc

```diff
@@ -290,7 +290,7 @@ std::string ignition::common::joinPaths(const std::string &_path1,
 
     // Sanitize the end of the path.
     index = result.length()-1;
-    for (; result[index] == replacement; --index)
+    for (; index <  result.length() && result[index] == replacement; --index)
```

Comment:
should we add logic like this on line 285 as well?

Reply:
I think the first loop is safe, but it probably wouldn't hurt to add.  Done in `b9e4cc0`